integration-tests/bats/sql-server.bats

```diff
@@ -1504,9 +1504,9 @@ databases:
     dolt sql-server --port=$PORT --socket="$REPO_NAME/mysql.sock" --user dolt > log.txt 2>&1 &
     SERVER_PID=$!
     run wait_for_connection $PORT 5000
-    [ "$status" -eq 1 ]
+    [ "$status" -eq 0 ]
 
-    run grep 'address already in use' log.txt
+    run grep 'unix socket set up failed: address already in use: listen unix at' log.txt
```

Comment:
Nice test for this! 

It might be nice to also test that when we shut down the second sql-server, it doesn't delete the `mysql.sock` file that it tried to create, but wasn't able to. I think this will happen correctly from the code changes, but probably worth a quick line or two to assert it behaves that way, too.